File: src/Models/Request/AttributeRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;
using Models.Enums;

namespace Models.Request;

[ExcludeFromCodeCoverage]
public class AttributeRequest()
{
    [Required(ErrorMessage = "Name is required.")]
    public string? Name { get; set; }

    [JsonIgnore]
    public Guid ReferenceId => Guid.TryParse(IdReference, out var guid) ? guid : Guid.Empty;

    [Required(ErrorMessage = "IdReference is required.")]
    public string IdReference { get; set; } = string.Empty;

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public SimModelsPrivacity Privacity { get; set; }

    [JsonRequired]
    public bool IsStatic { get; set; }
}

File: frontend/src/app/models/request/AttributeRequest.ts
export class AttributeRequest {
  id: string = '';
  name: string = '';
  idReference: string = '';
  idInstance: string = '';
  privacity: string = 'Public';
  idRelatedClass: string = '';
  isStatic: boolean = false;

  toString(): string {
    return `AttributeRequest(id=${this.id}, name=${this.name}, referenceId=${this.idReference}, instanceId=${this.idInstance}, privacity=${this.privacity}, relatedClassId=${this.idRelatedClass}, isStatic=${this.isStatic})`;
  }
}